sql/moz-fx-data-marketing-prod/ga_derived/downloads_with_attribution_v2/query.sql

```diff
@@ -0,0 +1,367 @@
+    mozfun.stats.mode_last_retain_nulls(ARRAY_AGG(medium)) AS medium,
+    mozfun.stats.mode_last_retain_nulls(ARRAY_AGG(source)) AS source,
+    mozfun.stats.mode_last_retain_nulls(ARRAY_AGG(device_category)) AS device_category,
+    mozfun.stats.mode_last_retain_nulls(ARRAY_AGG(os)) AS os,
+    mozfun.stats.mode_last_retain_nulls(ARRAY_AGG(browser)) AS browser,
+    mozfun.stats.mode_last_retain_nulls(ARRAY_AGG(browser_version)) AS browser_version,
+    mozfun.stats.mode_last_retain_nulls(ARRAY_AGG(`language`)) AS `language`,
+    s.stub_visit_id AS stub_visit_id,
+    s.stub_download_session_id AS stub_download_session_id,
+    s.dltoken AS dltoken,
+    mozfun.stats.mode_last_retain_nulls(ARRAY_AGG(landing_page)) AS landing_page,
+    mozfun.stats.mode_last_retain_nulls(ARRAY_AGG(pageviews)) AS pageviews,
+    mozfun.stats.mode_last_retain_nulls(ARRAY_AGG(unique_pageviews)) AS unique_pageviews,
+    LOGICAL_OR(
+      has_ga_download_event
+    ) AS has_ga_download_event,  -- this will be ignored if nrows >1
+    mozfun.stats.mode_last_retain_nulls(
+      ARRAY_AGG(count_dltoken_duplicates)
+    ) AS count_dltoken_duplicates,
+    LOGICAL_OR(additional_download_occurred) AS additional_download_occurred,
+    COUNT(*) AS nrows,
+    mozfun.stats.mode_last_retain_nulls(ARRAY_AGG(s.download_date)) AS download_date,
+    mozfun.stats.mode_last_retain_nulls(ARRAY_AGG(time_on_site)) AS time_on_site
+  FROM
+    ga_sessions_with_hits gs
+  RIGHT JOIN
+    stub s
+  ON
+    (gs.client_id = s.stub_visit_id AND gs.download_session_id = s.stub_download_session_id)
+  GROUP BY
+    gs.client_id,
+    dltoken,
+    stub_visit_id,
+    stub_download_session_id
+)
+SELECT
+  dltoken,
+  download_date,
+  CASE
+    WHEN nrows > 1
+      THEN NULL
+    ELSE time_on_site
+  END
+  time_on_site,
+  CASE
+    WHEN nrows > 1
+      THEN NULL
+    ELSE ad_content
+  END
+  ad_content,
+  CASE
+    WHEN nrows > 1
+      THEN NULL
+    ELSE campaign
+  END
+  campaign,
+  CASE
+    WHEN nrows > 1
+      THEN NULL
+    ELSE medium
+  END
+  medium,
+  CASE
+    WHEN nrows > 1
+      THEN NULL
+    ELSE source
+  END
+  source,
+  CASE
+    WHEN nrows > 1
+      THEN NULL
+    ELSE landing_page
+  END
+  landing_page,
+  CASE
+    WHEN nrows > 1
+      THEN NULL
+    ELSE country
+  END
+  country,
+  CASE
+    WHEN nrows > 1
+      THEN NULL
+    ELSE cn.code
+  END
+  normalized_country_code,
+  CASE
+    WHEN nrows > 1
+      THEN NULL
+    ELSE device_category
+  END
+  device_category,
+  CASE
+    WHEN nrows > 1
+      THEN NULL
+    ELSE os
+  END
+  os,
+  CASE
+    WHEN nrows > 1
+      THEN NULL
+    WHEN os IS NULL
+      THEN NULL
+    WHEN os LIKE 'Macintosh%'
+      THEN 'Mac'  -- these values are coming from GA.
+    ELSE mozfun.norm.os(os)
+  END
+  normalized_os,
+  CASE
+    WHEN nrows > 1
+      THEN NULL
+    ELSE browser
+  END
+  browser,
+  CASE
+    WHEN nrows > 1
+      THEN NULL
+    ELSE normalize_browser(browser)
```

Comment:
Can we normalize in the view, and keep the raw value here (then we can later add e.g. opera, brave)

Reply:
The raw value is maintained in the table as well.  There is no view currently available due to the changes in how the data is processed from v1 to v2.  The user of the table needs to know if the data is v1 or v2 to understand the results.